src/app/open-dialog.directive.ts

```diff
@@ -0,0 +1,22 @@
+import {Directive, HostListener, HostBinding} from '@angular/core';
+import {LoginDialogComponent} from './login-dialog/login-dialog.component';
```

Comment:
So the name of the directive implies, that it's a dynamic directive (you can open any type of dialog) but it's tightly coupled to login. Also, I think it's not in the right place. If you can make it dynamic (it can open the dialog that's passed down somehow as a parameter), then please create an `/arc/app/common` folder and put it there. If you keep it coupled to login (which is fine), please rename it to open-login-dialog and put it in session, where the rest of the auth stuff is.